relayer/middleware/source-tx.middleware.ts

```diff
@@ -106,41 +127,20 @@ export async function fetchVaaHash(
   emitterAddress: Buffer,
   sequence: bigint,
   logger: Logger,
-  env: Environment,
-  retries: number = 3,
-  baseEndpoint: string = wormscanEndpoints[env],
+  wormscan: WormscanClient,
 ) {
-  let attempt = 0;
-  let txHash = "";
-  do {
-    let body;
-    try {
-      const res = await fetch(
-        `${baseEndpoint}/api/v1/vaas/${emitterChain}/${emitterAddress.toString(
-          "hex",
-        )}/${sequence.toString()}`,
-      );
-      if (res.status === 404) {
-        throw new Error("Not found yet.");
-      }
+  const response = await wormscan.getVaa(
+    emitterChain,
+    emitterAddress.toString("hex"),
+    sequence,
+  );
 
-      // Note that we're assuming the response is UTF8 encoded here.
-      body = await res.text();
-      if (!res.ok) {
-        throw new Error(`Unexpected HTTP response. Status: ${res.status}`);
-      }
+  if (response.error) {
+    logger.error("Error fetching tx hash: " + response.error);
+    throw response.error;
+  }
 
-      // TODO: consider restricting this code path further to just status 200 and a few others.
-      txHash = JSON.parse(body).data?.txHash;
-    } catch (error) {
-      const httpBodyText =
-        body !== undefined ? `\nHTTP response body: ${body}` : "";
-      const errorMessage = `could not obtain txHash, attempt: ${attempt} of ${retries}.${httpBodyText}
-Error: ${error}`;
-      logger?.error(errorMessage);
-      await sleep((attempt + 1) * 200); // linear wait
-    }
-  } while (++attempt < retries && !txHash);
+  let txHash = response.data?.txHash || "";
```

Comment:
May be we can throw an error if we don't find `txHash` on the response data object and remove empty string as fallback.

Reply:
Maybe we should put a comment somewhere (or ideally on documentation if we had any), but...
It's a design decision not to throw an error. This is becase you wan't to leave this decision to the user of relayer-engine. The source tx hash is strictly necessary for processing some VAAs, but not for others, so it's up to the user to decide.
At some point we thought that maybe we can make this a configuration of the middleware but never got to implement it... for the time been I'd leave it as is